modules/linear_boltzmann_solvers/response_evaluator/response_evaluator.cc

```diff
@@ -0,0 +1,468 @@
+                         "Parameter \"group_strength\" is required for "
+                         "boundaries of type \"incident_isotropic\".");
+    params.RequireParameterBlockTypeIs("values", ParameterBlockType::ARRAY);
+
+    boundary_sources_[bid] = {BoundaryType::INCIDENT_ISOTROPIC,
+                              params.GetParamVectorValue<double>("group_strength")};
+  }
+  else
+    log.Log0Warning() << "Unsupported boundary type. Skipping the entry.";
+}
+
+void
+ResponseEvaluator::ClearForwardSources()
+{
+  material_sources_.clear();
+  point_sources_.clear();
+  distributed_sources_.clear();
+  boundary_sources_.clear();
+}
+
+double
+ResponseEvaluator::EvaluateResponse(const std::string& buffer) const
+{
+  const auto& buffer_data = adjoint_buffers_.at(buffer);
+  const auto& phi_dagger = buffer_data.first;
+  const auto& psi_dagger = buffer_data.second;
+
+  ChiLogicalErrorIf(not material_sources_.empty() and phi_dagger.empty(),
+                    "If material sources are present, adjoint flux moments "
+                    "must be available for response evaluation.");
+  ChiLogicalErrorIf(not point_sources_.empty() and phi_dagger.empty(),
+                    "If point sources are set, adjoint flux moments "
+                    "must be available for response evaluation.");
+  ChiLogicalErrorIf(not distributed_sources_.empty() and phi_dagger.empty(),
+                    "if distributed sources are set, adjoint flux moments "
+                    "must be available for response evaluation.");
+  ChiLogicalErrorIf(not boundary_sources_.empty() and psi_dagger.empty(),
+                    "If boundary sources are set, adjoint angular fluxes "
+                    "must be available for response evaluation.");
+
+  const auto& grid = lbs_solver_.Grid();
+  const auto& discretization = lbs_solver_.SpatialDiscretization();
+  const auto& transport_views = lbs_solver_.GetCellTransportViews();
+  const auto& unit_cell_matrices = lbs_solver_.GetUnitCellMatrices();
+  const auto num_groups = lbs_solver_.NumGroups();
+
+  double local_response = 0.0;
+
+  // Material sources
+  if (not material_sources_.empty())
+  {
+    for (const auto& cell : grid.local_cells)
+    {
+      const auto& cell_mapping = discretization.GetCellMapping(cell);
+      const auto& transport_view = transport_views[cell.local_id_];
+      const auto& fe_values = unit_cell_matrices[cell.local_id_];
+      const auto num_cell_nodes = cell_mapping.NumNodes();
+
+      if (material_sources_.count(cell.material_id_) > 0)
+      {
+        const auto& src = material_sources_.at(cell.material_id_);
+        for (size_t i = 0; i < num_cell_nodes; ++i)
+        {
+          const auto dof_map = transport_view.MapDOF(i, 0, 0);
+          const auto& V_i = fe_values.intV_shapeI[i];
+          for (size_t g = 0; g < num_groups; ++g)
+            local_response += src[g] * phi_dagger[dof_map + g] * V_i;
+        }
+      }
+    } // for cell
+  }   // if material sources
```

Comment:
should you remove 2 white spaces here to move the comment a tad to the left?

Reply:
This is what our current clang-format does. If I change it I probably fail the checks. As far as I know, this is how it is within the rest of the code base.